dht/token.hh

```diff
@@ -98,10 +98,15 @@ public:
     }
 
     bytes data() const {
-        auto t = net::hton(_data);
-        bytes b(bytes::initialized_later(), sizeof(_data));
-        std::copy_n(reinterpret_cast<int8_t*>(&t), sizeof(_data), b.begin());
-        return b;
+        return write_bytes_be(_data);
+    }
+
+    bytes unsigned_data() const {
+        // Map range from [-2^63, 2^63-1] to [0, 2^64-1]. Written this way 
+        // (subtracting int64_t min) to avoid C++ undefined behaviour.
```

Comment:
This comment is a bit confusing. Can't we just add (1ull << 63) to unsigned_value?